herbert_core/utilities/geometry/box_intersect.m

```diff
@@ -0,0 +1,156 @@
+function inter_points = box_intersect(box_minmax,cross_plain)
+% Calculate intersection points between the box in ND (2,3,4)
+% and line/plain/hyperplane of dimension (N-1)D
+%
+% Inputs:
+% box_minmax -- NDx2 array of min and max points of the box, to intersect with.
+%
+% cross_plain -- NDxND array of points defining plain in the appropriate
+%                dimensions. The coordinates defined along the first
+%                dimension and the second dimention correspont to number of
+%                points, the plain is passing through (has to be ND). If
+%                there are ND-1 points, the missing point assumed to be
+%                equal to 0.
```

Comment:
So is this a grid aligned with the q and energy axes, or a general one. Are the points defined as indices in some array or as 4D coordinates (xyzt)? Is it a general curvilinear lattice or an orthogonal linear one? If the coordinates have been all separately specified, why is a separate number for the number of points? I think an example would help make this clear.

Reply:
The grit is general orthogonal one (how projection defines it)  and the box is aligned with this coordinate system.
Its loosely related to generic projection but will be used by it. This will work for bounding boxes or lattices, where curvature is ignored (edges are strait lines). At the moment I am using it to calculate ranges of symetrisation in a non-cubic lattice. 

 The efficiency may need to be improved if we want to calculate intersections of two grids.  (see symmetrise_horace_2d)